features/steps/gradle_plugin_steps.rb

```diff
@@ -54,13 +54,10 @@
 end
 
 When("I build the failing {string} using the {string} bugsnag config") do |module_config, bugsnag_config|
-  begin
-    steps %Q{
-      When I build "#{module_config}" using the "#{bugsnag_config}" bugsnag config
-    }
-    assert(false, "Expected script to fail with non-zero exit code")
-  rescue SystemExit
-  end
+  Runner.environment["MODULE_CONFIG"] = module_config
+  Runner.environment["BUGSNAG_CONFIG"] = bugsnag_config
+  _, exit_code = Runner.run_script("features/scripts/bundle_project_module.sh", blocking: true)
```

Comment:
I'm not overly fond of this re-write, but can't think of a better way of getting the exit_code.  It's using the tools we've previously written, but just feels off.

Reply:
Return the exit code from the method that invoked it seems perfectly reasonable to me.